tests/integration_tests/utils.py

```diff
@@ -520,7 +520,18 @@ def run_api_experiment(input_features, output_features, data_csv):
         shutil.rmtree(output_dir, ignore_errors=True)
 
 
-def create_data_set_to_use(data_format, raw_data):
+def read_csv_with_nan(path, nan_percent=0.0):
```

Comment:
This function converts `nan_percent` of samples in each row of the CSV into NaN. This is important for tests that drop rows for exactly one feature– with this change, one has guarantees about the number of rows that will be dropped.

Example: 
We are predicting over `targets` column which is missing